xskillscore/core/resampling.py

```diff
@@ -0,0 +1,211 @@
+import dask
+import numpy as np
+import xarray as xr
+
+CONCAT_KWARGS = {"coords": "minimal", "compat": "override"}
+
+
+def _gen_idx(forecast, dim, iterations, select_dim_items, replace, new_dim):
+    """Generate indices to select from. replace decides whether resampling is with or
+    without replacement."""
+    if replace:
+        idx = np.random.randint(0, forecast[dim].size, (iterations, select_dim_items))
+    elif not replace:
+        # create 2d np.arange
+        idx = np.linspace(
+            (np.arange(select_dim_items)),
+            (np.arange(select_dim_items)),
+            iterations,
+            dtype="int",
+        )
+        # shuffle each line
+        for ndx in np.arange(iterations):
+            np.random.shuffle(idx[ndx])
+    idx_da = xr.DataArray(
+        idx,
+        dims=("iteration", dim),
+        coords=({"iteration": range(iterations), dim: new_dim}),
+    )
+    return idx_da
+
+
+def resample_iterations(forecast, iterations, dim="member", dim_max=None, replace=True):
+    """Resample over ``dim`` by index ``iterations`` times.
+
+    .. note::
+        This gives the same result as
+        :py:func:`~xskillscore.resampling.resample_iterations_idx` but slower.
+        When using ``dask``, the number of tasks in
+        :py:func:`~xskillscore.resampling.resample_iterations` will scale with
+        ``iterations`` but constant chunksize, whereas the tasks in
+        :py:func:`~xskillscore.resampling.resample_iterations_idx` will stay constant
+        with increasing chunksize.
+
+    Parameters
+    ----------
+    forecast : xr.DataArray, xr.Dataset)
+        forecast.
+    iterations :int
+        Number of resampling iterations.
+    dim : str
+        Dimension name to resamplw over. Defaults to ``'member'``.
```

Comment:
Typo on resample

Reply:
Should we have a default here?

Reply:
I think its ok to have a default here. we also have that default for member_dim